lib/ecto/repo.ex

```diff
@@ -1828,43 +1828,51 @@ defmodule Ecto.Repo do
-  abort the transaction. For example, if you attempt an insert that violates
-  a unique constraint, the insert fails and the transaction is aborted.
-  In such cases, any further operation inside the transaction will raise
-  exceptions.
+  When an operation inside a transaction fails, the transaction is aborted in the database. 
+  For instance, if you attempt an insert that violates a unique constraint, the insert fails 
+  and the transaction is aborted. In such cases, any further operation inside the transaction 
+  will raise exceptions.
 
-  Take the following transaction:
+  Take the following transaction as an example:
 
       Repo.transaction(fn repo ->
         case repo.insert(changeset) do
           {:ok, post} ->
-            post
+            repo.insert(%Success{})
 
           {:error, changeset} ->
             repo.insert(%Failure{})
         end
       end)
 
-  If the changeset is invalid before it reaches the database,
-  for example due to a validation error, no statement is sent
-  to the database, an `:error` tuple is returned, and the
-  `repo.insert(%Failure{})` operation will execute as usual.
-  However, if the changeset is valid but the insert operation
-  fails due to a database constraint, the subsequent
-  `repo.insert(%Failure{})` operation will raise an exception
-  because the database has aborted the transaction and made any
-  subsequent operation invalid. In Postgres, it would look like
-  this:
-
+  If the changeset is valid, but the insert operation fails due to a database constraint,
+  the subsequent `repo.insert(%Failure{})` operation will raise an exception because the
+  database has already aborted the transaction and thus making the operation invalid.
+  In Postgres, the exception would look like this:
+  
       ** (Postgrex.Error) ERROR 25P02 (in_failed_sql_transaction) current transaction is aborted, commands ignored until end of transaction block
-
-  Such scenarios must be handled outside of the transaction.
-  For example, you can choose to perform an explicit `repo.rollback`
-  call in the `{:error, changeset}` clause and then perform the
-  `repo.insert(%Failure{})` outside of the transaction. Alternatively,
-  consider using `Ecto.Multi`, as they automatically rollback whenever
-  an operation fails.
+      
+  If the changeset is invalid before it reaches the database due to a validation error, 
+  no statement is sent to the database, an `:error` tuple is returned, and `repo.insert(%Failure{})` 
+  operation will execute as usual. 
+
+  We have a few options to deal with such scenarios:
+  
+  The most obvious one would be to prevent the the database from raising using 
+  something like `repo.insert(post, on_conflict: :nothing)`. In this case, the 
+  operation will still fail at the database level, but since we are supressing the error, 
+  we will receive back a `:ok` tuple with an unsaved post: `%Post{id: nil}`.
```

Comment:
This is actually much more complex because `on_conflict: :nothing` is database specific and you can match on `id: nil` only if you are using auto generated primary keys and it also means the error is no longer added to the changeset. I am thinking we should not recommend this path:

```suggestion
  We have two options to deal with such scenarios:
```

Reply:
Don’t you think that we should at least present this as an option with some remarks about it? I’d rather making clear it’s not recommended than denying the information (I’m surprised how this didn’t come up more often in the forums btw).

I’m coming from the perspective of giving sharp tools to users and let them decide. 

I trust your judgment though if you think this absolutely won’t improve anything. :)